ratis-common/src/main/java/org/apache/ratis/util/PeerProxyMap.java

```diff
@@ -165,8 +165,10 @@ public PROXY createProxyImpl(RaftPeer peer) throws IOException {
 
   @Override
   public void close() {
-    peers.values().parallelStream().forEach(
-        pp -> pp.setNullProxyAndClose().ifPresent(proxy -> closeProxy(proxy, pp)));
+    ConcurrentUtils.parallelForEachAsync(peers.values(),
+        pp -> pp.setNullProxyAndClose().ifPresent(proxy -> closeProxy(proxy, pp)),
+        r -> new Thread(r).start()
+    );
```

Comment:
@adoroszlai , thanks for working on this!  We should join the returned future in order to preserve the behavior.
```java
    ).join();
```